File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    result = 1
    while(result <= 500):
        if(num == 1):
            result = 0
            break
        if num % 2 == 0:
            num /= 2
        else:
            num = (num * 3) + 1
        
        if(num == 1):
            break
            
        if(result >= 500 and num != 1):
            result = -1
            break
            
        result += 1
    return result

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        int tmp = 0;
        if(num == 1) return 0;
        long ans = (long) num;
        while(tmp < 500) {
            if(ans % 2 == 0) ans /= 2;
            else ans = ans*3 + 1;
            tmp++;
            if(ans == 1) return tmp;
        }
        return -1;
    }
}